entrypoint/internal/entrypoint/entrypoint.go

```diff
@@ -0,0 +1,327 @@
+	"github.com/Mellanox/doca-driver-build/entrypoint/internal/constants"
+	"github.com/Mellanox/doca-driver-build/entrypoint/internal/driver"
+	"github.com/Mellanox/doca-driver-build/entrypoint/internal/netconfig"
+	"github.com/Mellanox/doca-driver-build/entrypoint/internal/utils/cmd"
+	"github.com/Mellanox/doca-driver-build/entrypoint/internal/utils/host"
+	"github.com/Mellanox/doca-driver-build/entrypoint/internal/utils/ready"
+	"github.com/Mellanox/doca-driver-build/entrypoint/internal/utils/udev"
+	"github.com/Mellanox/doca-driver-build/entrypoint/internal/wrappers"
+)
+
+// Start the entrypoint manager with file-based locking to ensure only one instance runs at a time.
+// Handlers in the entrypoint manager:
+//   - preStart: Cleans up, validates, and prepares. If it fails,
+//     the process exits immediately without running "stop".
+//   - start: Builds and loads the driver after preStart succeeds. If successful,
+//     the manager waits for a termination signal. If it fails, "stop" still runs.
+//   - stop: Handles unloading the driver and container teardown.
+func Run(signalCh chan os.Signal, log logr.Logger, containerMode string, cfg config.Config) error {
+	osWrapper := wrappers.NewOS()
+	cmdHelper := cmd.New()
+	hostHelper := host.New(cmdHelper, osWrapper)
+	m := &entrypoint{
+		log:           log,
+		config:        cfg,
+		containerMode: containerMode,
+		readiness:     ready.New(cfg.DriverReadyPath, osWrapper),
+		udev:          udev.New(cfg.MlxUdevRulesFile, osWrapper),
+		host:          hostHelper,
+		cmd:           cmdHelper,
+		os:            osWrapper,
+		netconfig:     netconfig.New(),
+		drivermgr:     driver.New(containerMode, cfg, cmdHelper, hostHelper, osWrapper),
+	}
+	return m.run(signalCh)
+}
+
+// entrypoint orchestrates the high-level logic for loading and unloading the driver.
+type entrypoint struct {
+	log logr.Logger
+
+	config        config.Config
+	containerMode string
+
+	drivermgr driver.Interface
+	netconfig netconfig.Interface
+	cmd       cmd.Interface
+	readiness ready.Interface
+	udev      udev.Interface
+	os        wrappers.OSWrapper
+	host      host.Interface
+}
+
+// run is an actual implementation of the entrypoint.Run()
+func (e *entrypoint) run(signalCh chan os.Signal) error {
+	unlock, err := e.lock()
+	if err != nil {
+		return err
+	}
+	defer unlock()
+
+	startCtx, startCancel := context.WithCancel(context.Background())
+	stopCtx, stopCancel := context.WithCancel(context.Background())
+	startCtx = logr.NewContext(startCtx, e.log)
+	stopCtx = logr.NewContext(stopCtx, e.log)
+	setupSignalHandler(signalCh, []ctxData{{Ctx: startCtx, Cancel: startCancel}, {Ctx: stopCtx, Cancel: stopCancel}})
+
+	e.log.Info("NVIDIA driver container exec preStart")
+	if err := e.preStart(startCtx); err != nil {
+		e.log.Error(err, "exec preStart failed")
+		return err
+	}
+	e.log.Info("NVIDIA driver container exec start")
+	startErr := e.start(startCtx)
+	if startErr != nil {
+		e.log.Error(err, "exec start failed")
```

Comment:
Shouldn't we return `startErr` here
```suggestion
		e.log.Error(startErr, "exec start failed")
```